syntax = "proto3";

package convertidor_de_archivos;

message ErrorGeneral {
  string error = 1;
  string mensaje = 2;
}

enum CalidadCancion{
  ALTA = 0;
  MEDIA = 1;
  BAJA = 2;
}

message InformacionArchivo {
  int64 idCancion = 1;
  string extension = 2;
  string hash256 = 3;
}

message Chunk {
  bytes data = 1;
}

message SolicitudConvertirCancionMp3 {
  InformacionArchivo informacionArchivo = 1;
  Chunk paquete = 2;
}

message RespuestaCancionConvertida {
  InformacionArchivo informacionArchivo = 1;
  Chunk paquete = 2;
}

message RespuestaCancionesConvertidas {
  RespuestaCancionConvertida cancionCalidadAlta = 1;
  RespuestaCancionConvertida cancionCalidadMedia = 2;
  RespuestaCancionConvertida cancionCalidadBaja = 3;
  ErrorGeneral error = 4;
}

service ConvertidorDeCanciones{
  rpc ConvertirCancionAMp3(stream SolicitudConvertirCancionMp3) returns (stream RespuestaCancionesConvertidas) {};
}

